Origin: upstream, https://github.com/cyrusimap/cyrus-sasl/pull/644/
Bug-Debian: https://bugs.debian.org/1003355
Description: Fix configure.ac for autoconf 2.70
---
Ubuntu note:
 The 3 individual commits that make up this diff are:
 https://github.com/cyrusimap/cyrus-sasl/pull/644/commits/999255e05719f61bdbce8125be2ee774493aa64a
 https://github.com/cyrusimap/cyrus-sasl/pull/644/commits/15cbc14aeb4b754b1b3db65f7c892c7deabaab41
 https://github.com/cyrusimap/cyrus-sasl/pull/644/commits/5664c3f535289ce9efb513a2897991b5c436bb44
diff --git a/configure.ac b/configure.ac
index 388f5d02..251e43b5 100644
--- a/configure.ac
+++ b/configure.ac
@@ -261,18 +261,23 @@ if test $sasl_cv_uscore = yes; then
 	AC_MSG_CHECKING(whether dlsym adds the underscore for us)
 	cmu_save_LIBS="$LIBS"
 	LIBS="$LIBS $SASL_DL_LIB"
-	AC_CACHE_VAL(sasl_cv_dlsym_adds_uscore,AC_TRY_RUN( [
+	AC_CACHE_VAL([sasl_cv_dlsym_adds_uscore],
+	             [AC_TRY_RUN([
 #include <dlfcn.h>
 #include <stdio.h>
+#include <stdlib.h>
 void foo() { int i=0;}
 int main() { void *self, *ptr1, *ptr2; self=dlopen(NULL,RTLD_LAZY);
     if(self) { ptr1=dlsym(self,"foo"); ptr2=dlsym(self,"_foo");
     if(ptr1 && !ptr2) exit(0); } exit(1); } 
-], [sasl_cv_dlsym_adds_uscore=yes], sasl_cv_dlsym_adds_uscore=no
-	AC_DEFINE(DLSYM_NEEDS_UNDERSCORE, [], [Do we need a leading _ for dlsym?]),
-	AC_MSG_WARN(cross-compiler, we'll do our best)))
+], [sasl_cv_dlsym_adds_uscore=yes], [sasl_cv_dlsym_adds_uscore=no],
+	[AC_MSG_WARN(cross-compiler, we'll do our best)])])
 	LIBS="$cmu_save_LIBS"
       AC_MSG_RESULT($sasl_cv_dlsym_adds_uscore)
+
+      if test "$sasl_cv_dlsym_adds_uscore" = no; then
+        AC_DEFINE(DLSYM_NEEDS_UNDERSCORE, [], [Do we need a leading _ for dlsym?])
+      fi
   fi
 fi
 
